src/lib/hooks/swap/useSendSwapTransaction.tsx

```diff
@@ -91,15 +94,18 @@ export default function useSendSwapTransaction({
                 `Your swap was modified through your wallet. If this was a mistake, please cancel immediately or risk losing your funds.`
               )
             }
+            analytics.track(AnalyticsEvent.SWAP_WALLET_CONFIRM_TRANSACTION_CLICKED)
             return response
           })
           .catch((error) => {
             // if the user rejected the tx, pass this along
-            if (error?.code === 4001) {
+            if (error?.code === 4001 || error.reason === 'user rejected transaction') {
```

Comment:
the `reason` here is exactly this? wouldn't be safer to do:

```suggestion
            if (error?.code === 4001 || error.reason.toLowerCase().includes('user rejected transaction')) {
```

Reply:
Nice catch 👍